### PR TITLE
Add AudioToText and AudioToTextPreprocessor class stubs to enable auto class functionality

### PR DESCRIPTION
**Pull request by harshaljanjani**
## Description of the change

This PR was originally intended as a design diff to make MoonshineAudioConverter and WhisperAudioConverter independent of tf ops and leave the tf-specific handling to the AudioConverter base class. **However, it is now restricted to adding class stubs for auto class functionality.** Please review the following considerations that should be kept in mind for the design diff regardless.

We have two choices in the matter:

a. Make AudioConverter **a layer with specific arguments**, an init(), call() and get_config().

I believe this option is a bit too strict given that at the moment, in the art at present, we have two classes with highly divergent parameters in themselves. I believe the preprocessing fundamentally differs from the other, and having an argument such as `use_spectrogram_features` might be a bit too premature, since in Whisper itself, it uses a specific variant, log-mel spectrogram features with Whisper-specific audio preprocessing.

1. **Moonshine:** The input is a waveform, and the output is a tensor for an audio waveform as well. There's manipulation in terms of padding and normalization but effectively no feature extraction with STFTs like Whisper, etc.
2. **Whisper:** Raw waveform to spectrogram, so it's time → time-frequency domain.

b. The choice this PR is based on makes AudioConverter a **core ops class** that future AudioConverters can subclass from. These compatibility issues will be ubiquitous when writing the AudioConverter and would have to be dealt with when using the tf-specific ops; instead, they can use the base class ops without worrying too much about the tf-specific handling.
I've taken efforts here to eliminate the need for the developer to worry about the low-level utils they share. Also, this itself would be a great addition because it'll clean up a lot of messy backend-specific work in the audio converters, in my opinion!

This cleans up MoonshineAudioConverter and WhisperAudioConverter.

## Checklist

- [x] I have added all the necessary unit tests for my change (here, the fact that none of the tests break is test enough to the fact that the logic is sound).
- [x] I have verified that my change does not break existing code and works with all backends (TensorFlow, JAX, and PyTorch).
- [x] My PR is based on the latest changes of the main branch (if unsure, rebase the code).
- [x] I have followed the Keras Hub [Model contribution guidelines](https://github.com/keras-team/keras-hub/blob/master/CONTRIBUTING_MODELS.md) in making these changes.
- [x] I have followed the Keras Hub [API design guidelines](https://github.com/keras-team/keras-hub/blob/master/API_DESIGN_GUIDE.md) in making these changes.
- [x] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

**Comment by harshaljanjani** [comment]
@mattdangerw / @divyashreepathihalli CI is green on the cleanup refactor! Once you approve of this, I'll proceed to add the class stubs **within the scope of this PR itself**. Since it's a subtle design change, it shouldn't break CI, you may just review the diff right after, thanks!

File: keras_hub/src/layers/preprocessing/audio_converter.py [hunk 1]
```diff
@@ -102,3 +106,216 @@ def save_to_preset(self, preset_dir):
...
+            if mode == "constant":
+                return keras.ops.pad(
+                    tensor, paddings, mode=mode, constant_values=constant_values
+                )
+            else:
+                return keras.ops.pad(tensor, paddings, mode=mode)
+
+    def _cond(self, condition, true_fn, false_fn, tensor):
+        """Conditional execution based on backend and tensor type."""
+        if self._use_tf_graph_ops(tensor) and keras.config.backend() != "torch":
```
[the remark this answers is not in the document]
Comment (harshaljanjani) [reply keras_hub/src/layers/preprocessing/audio_converter.py:157]: This is something I thought I learnt from the Moonshine PR. In the Moonshine PR I faced a CI issue with Torch-GPU and I fixed it like this:
https://github.com/keras-team/keras-hub/pull/2093/commits/92dc884cdcf274d5beb09e9502645467725551c8, involved a conditional check for the Torch backend for `tf.cond`.

File: keras_hub/src/layers/preprocessing/audio_converter.py [hunk 2]
```diff
@@ -102,3 +106,216 @@ def save_to_preset(self, preset_dir):
...
+        compress=False,
+        compression_threshold=8.0,
+    ):
+        """Compute the log-mel spectrogram."""
+        if tf is None:
+            raise ImportError(
+                "TensorFlow is required for computing the log mel spectrogram."
+            )
+        if isinstance(audio, torch.Tensor):
+            audio = audio.cpu()
```
[the remark this answers is not in the document]
Comment (harshaljanjani) [reply keras_hub/src/layers/preprocessing/audio_converter.py:240]: I see, I'll remove the redundancy after verifying breakage, thanks!

**Comment by harshaljanjani** [comment]
Thank you for the reviews @mattdangerw! For now, the scope of this PR is limited to the class stubs for auto class functionality, as you mentioned, and it is merge-ready. However, we can address the AudioConverter after a thorough design discussion with all of you!

**Comment by harshaljanjani** [comment]
Thank you for the reviews @mattdangerw! For now, the scope of this PR is limited to the class stubs for auto class functionality, as you mentioned. However, we can address the AudioConverter in the near future after a thorough design discussion with all of you, thanks!

**Review by mattdangerw — APPROVED** [review]
Nice! This looks great
